build.sh

```diff
@@ -445,6 +447,14 @@ if (( ${NUMARGS} == 0 )) || hasArg rust; then
     cargo test
 fi
 
+# Build the cuvs Java bindings
+if (( ${NUMARGS} == 0 )) || hasArg java; then
+    # build libcuvs first as the Java API depends on it
+    ./$0 libcuvs
```

Comment:
I'm not sure if it is good practice to call this script recursively to build the C++ code. It seems like it will lead to unexpected behavior. For example, how do you ensure that the CLI arguments such as `--allgpuarch` or `--no-mg` are being passed through?

I would recommend removing this line, and users must call `./build.sh libcuvs java` if the C++ has not yet been built. Perhaps a warning can be issued if `hasArg libcuvs` is false. I believe this is how we handle the Python builds, which also depend on C++ libraries being built.
```suggestion
```